### DISCUSSION TITLE
Getting NaN while using The ButterWorth Band Pass Filter

### DISCUSSION THREAD

**Discussion by Richard-Quinlivan** [ar1st0crat/NWaves#47] [Q&A] [answered]
I'm trying to use the ButterWorth filters to make a band pass filter. But for some reason I keep getting NaN as the result. I've tried it with the Process() function doing it float by float (this is the first implementation below) and all at once with the ApplyTo() method on a Signal (the second Implementation below). Both give me a list of NaN. I'm not sure what I'm doing wrong.

![filter](https://user-images.githubusercontent.com/42681238/113955854-7f5ee900-97ea-11eb-9ea7-057230657a05.png)

![filter2](https://user-images.githubusercontent.com/42681238/113957139-b504d180-97ec-11eb-81b0-3a0b05b70af0.png)

**Comment by ar1st0crat** [comment, marked as answer]
Hi!

Frequencies are normalized onto [0, 0.5] range. So you need to divide them by ```fs``` (not by ```fs/2```). [See here](https://github.com/ar1st0crat/NWaves/wiki/Notes-for-non~experts-in-DSP#how-to-setup-my-filter-and-get-it-working).

If you worked with sciPy/MATLAB before, you'll also probably find [this page](https://github.com/ar1st0crat/NWaves/wiki/NWaves-for-MATLAB-and-sciPy-users#filter-design) useful.

Hope it helps.

**Reply by Richard-Quinlivan** [reply]
Ah I see, that worked. That was driving me nuts.
Thank you so much
